### PR TITLE
feat(lists-endpoint): Support for receiving a list child node from DSP-API (DSP-1385)

### PR DESCRIPTION
**Pull request by mdelez**
resolves DSP-1385

### CODE REVIEW

**Comment by mdelez** [comment]
@tobiasschweizer I'll discuss this during the meeting today

File: src/api/admin/lists/lists-endpoint-admin.ts [hunk 1]
```diff
@@ -232,10 +234,16 @@ export class ListsEndpointAdmin extends Endpoint {
      * 
      * @param listItemIri The IRI of the list.
      */
-    getList(listItemIri: string): Observable<ApiResponseData<ListResponse> | ApiResponseError> {
+    getList(listItemIri: string): Observable<ApiResponseData<ListResponse | ListChildNodeResponse> | ApiResponseError> {
```
Comment (tobiasschweizer) [inline src/api/admin/lists/lists-endpoint-admin.ts:237]: Will this be regarded a breaking change?

Comment (tobiasschweizer) [reply src/api/admin/lists/lists-endpoint-admin.ts:237]: Not sure if code actively using `getList` already will still compile.

File: src/api/admin/lists/lists-endpoint.spec.ts [hunk 2]
```diff
@@ -556,11 +557,11 @@ describe("ListsEndpoint", () => {
...
     describe("Method getList", () => {
 
-        it("should return a list", done => {
+        it("should return a list (root node)", done => {
 
             knoraApiConnection.admin.listsEndpoint.getList("http://rdfh.ch/lists/0001/treeList").subscribe(
-                (res: ApiResponseData<ListResponse>) => {
-                    expect(res.body.list.listinfo.id).toEqual("http://rdfh.ch/lists/0001/treeList");
+                (res: ApiResponseData<ListResponse | ListChildNodeResponse>) => {
+                    expect((res.body as ListResponse).list.listinfo.id).toEqual("http://rdfh.ch/lists/0001/treeList");
```
Comment (tobiasschweizer) [inline src/api/admin/lists/lists-endpoint.spec.ts:564]: I think without the assertion ` as ListResponse` this would not compile, right?

Comment (mdelez) [reply src/api/admin/lists/lists-endpoint.spec.ts]: It seems to work without the casts as well. I removed them in 0e40bb2

File: src/api/admin/lists/lists-endpoint-admin.ts (hunk 1 above)
Comment (mdelez) [reply src/api/admin/lists/lists-endpoint-admin.ts:237]: I think if you only specify one return type, then it shouldn't break anything. I'm actually doing that in [this file ](https://github.com/dasch-swiss/dsp-app/blob/8ed2d9b0d61b64c0cde83d227b13c7614470e976/src/app/project/list/list-item/list-item.component.ts#L47-L51) and [below in the same file](https://github.com/dasch-swiss/dsp-app/blob/8ed2d9b0d61b64c0cde83d227b13c7614470e976/src/app/project/list/list-item/list-item.component.ts#L102-L111) I use the two different return types (currently testing with yalc)

**Review by tobiasschweizer — APPROVED** [review]

**Pull request merged**